ga4gh/datamodel/sequenceAnnotations.py

```diff
@@ -0,0 +1,301 @@
+
+"""
+    For this implementation, `featureSetId` is required, while `parentId`
+    is optional, and filters the features within the requested `featureSetId`
+    by their parent.
+
+    Only return features on the reference with this name. Genomic positions
+    are non-negative integers less than reference length.
+    Requests spanning the join of circular genomes are represented as two
+    requests one on each side of the join (position 0) end is also required
+    If specified, this query matches only annotations which match one of the
+    provided feature types.
+    For now do not use the features array in search
+
+    GFF3 data is represented by rows in a single table, named FEATURE.
+    The columns of the FEATURE table correspond to the columns of a GFF3,
+    with three additional columns prepended representing the ID
+    of this feature, the ID of its parent (if any), and a whitespace
+    separated array of its child IDs.
+
+    _featureColumns pairs represent the ordered (column_name, column_type).
+"""
+_featureColumns = [('id', 'TEXT'),
+                   ('parent_id', 'TEXT'),
+                   ('child_ids', 'TEXT'),
+                   ('reference_name', 'TEXT'),
+                   ('source', 'TEXT'),
+                   ('ontology_term', 'TEXT'),
+                   ('start', 'INT'),
+                   ('end', 'INT'),
+                   ('score', 'REAL'),
+                   ('strand', 'TEXT'),  # limited to one of '+'/'-' or none.
+                   ('attributes', 'TEXT')  # JSON encoding of attributes dict
+                   ]
+
+
+class Gff3DbBackend(sqliteBackend.SqliteBackedDataSource):
+    """
+    Notes about the current implementation:
+    For this implementation, `featureSetId` is required, while `parentId`
+    is optional, and filters the features within the requested `featureSetId`
+    by their parent.
+
+    Genomic positions are non-negative integers less than reference length.
+    Requests spanning the join of circular genomes are represented as two
+    requests one on each side of the join (position 0)
+    """
+
+    def __init__(self, dbFile):
+        super(Gff3DbBackend, self).__init__(dbFile)
+        self.featureColumnNames = [f[0] for f in _featureColumns]
+        self.featureColumnTypes = [f[1] for f in _featureColumns]
+
+    def countFeaturesSearchInDb(self,
+                                referenceName=None, start=0, end=0,
+                                parentId=None, ontologyTerms=[]):
+        """
+        Same parameters as searchFeaturesInDb,
+        except without the pagetoken/size.
+        """
+        sql = ("SELECT COUNT(*) FROM FEATURE WHERE "
+               "reference_name = ? "
+               "AND end > ? "  # compare this to query start
+               "AND start < ? "  # and this to query end
+               )
+        sql_args = (referenceName, start, end)
+        if parentId is not None:
+            sql += "AND parent_id = ? "
```

Comment:
There's got to be a better way of doing this than string concatenation.   Do this instead: put formatting arguments in the `sql` string and set them to empty strings if not applicable.  e.g.

```
limit = ''
sql = 'SELECT * FROM Table {}'
if condition:
    limit = 'LIMIT 10'
finalSql = sql.format(limit)
```